web/web.go

```diff
@@ -0,0 +1,773 @@
+		sp.ColDefs[newName] = ddl.ColumnDef{
+			Name:    newName,
+			T:       sp.ColDefs[colName].T,
+			NotNull: sp.ColDefs[colName].NotNull,
+			Comment: sp.ColDefs[colName].Comment,
+		}
+		delete(sp.ColDefs, colName)
+	}
+	for i, pk := range sp.Pks {
+		if pk.Col == colName {
+			sp.Pks[i].Col = newName
+			break
+		}
+	}
+	srcColName := app.conv.ToSource[table].Cols[colName]
+	app.conv.ToSpanner[srcTableName].Cols[srcColName] = newName
+	app.conv.ToSource[table].Cols[newName] = srcColName
+	delete(app.conv.ToSource[table].Cols, colName)
+	app.conv.SpSchema[table] = sp
+}
+
+func pkChanged(pkChange, table, colName string) {
+	sp := app.conv.SpSchema[table]
+	if pkChange == "REMOVED" {
+		for i, pk := range sp.Pks {
+			if pk.Col == colName {
+				sp.Pks = removePk(sp.Pks, i)
+				break
+			}
+		}
+	}
+	if pkChange == "ADDED" {
+		if sPk, found := app.conv.SyntheticPKeys[table]; found {
+			for i, col := range sp.ColNames {
+				if col == sPk.Col {
+					sp.ColNames = remove(sp.ColNames, i)
+					break
+				}
+			}
+			delete(sp.ColDefs, sPk.Col)
+			for i, pk := range sp.Pks {
+				if pk.Col == sPk.Col {
+					sp.Pks = removePk(sp.Pks, i)
+					break
+				}
+			}
+		}
+		sp.Pks = append(sp.Pks, ddl.IndexKey{Col: colName, Desc: false})
+	}
+	app.conv.SpSchema[table] = sp
+}
+
+func updateType(newType, table, newColName, srcTableName string, w http.ResponseWriter) {
+	sp := app.conv.SpSchema[table]
+	srcColName := app.conv.ToSource[table].Cols[newColName]
+	srcCol := app.conv.SrcSchema[srcTableName].ColDefs[srcColName]
+	var ty ddl.Type
+	var issues []internal.SchemaIssue
+	switch app.driver {
+	case "mysql", "mysqldump":
+		ty, issues = toSpannerTypeMySQL(srcCol.Type.Name, newType, srcCol.Type.Mods)
+	case "pg_dump", "postgres":
+		ty, issues = toSpannerTypePostgres(srcCol.Type.Name, newType, srcCol.Type.Mods)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if len(srcCol.Type.ArrayBounds) > 1 {
+		ty = ddl.Type{Name: ddl.String, Len: ddl.MaxLength}
+		issues = append(issues, internal.MultiDimensionalArray)
+	}
+	if srcCol.Ignored.Default {
+		issues = append(issues, internal.DefaultValue)
+	}
+	if srcCol.Ignored.AutoIncrement {
+		issues = append(issues, internal.AutoIncrement)
+	}
+	if len(issues) > 0 {
+		app.conv.Issues[srcTableName][srcCol.Name] = issues
+	}
+	ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+	colDef := sp.ColDefs[newColName]
+	colDef.T = ty
+	sp.ColDefs[newColName] = colDef
+	app.conv.SpSchema[table] = sp
+}
+
+func updateNotNull(notNullChange, table, newColName string) {
+	sp := app.conv.SpSchema[table]
+	switch notNullChange {
+	case "ADDED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = true
+		sp.ColDefs[newColName] = spColDef
+	case "REMOVED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = false
+		sp.ColDefs[newColName] = spColDef
+	default:
+		// We skip this.
+	}
+	app.conv.SpSchema[table] = sp
```

Comment:
I don't think we need this last statement since we only change sp.ColDefs, and that is a map.

Reply:
removed last statement